Add visual styles for saas offerings

cshinn opened the pull request:
<img width="922" alt="screen shot 2017-04-04 at 4 18 52 pm" src="https://cloud.githubusercontent.com/assets/3300756/24676821/89bcf4de-1952-11e7-8bc4-fecce8c94054.png">
<img width="504" alt="screen shot 2017-04-04 at 4 19 23 pm" src="https://cloud.githubusercontent.com/assets/3300756/24676823/8aea2f16-1952-11e7-8626-9471b4cb34cf.png">

dtaylor113 commented:
Travis failure.  Think you need to run 'npm run build' and then push the dist files.

dtaylor113 commented:
When not hovering over a SaaS tile I cannot see a border for the tiles, or a change in color which separates the tiles from the background.  Not sure if that is intentional or not.

cshinn commented:
@dtaylor113 the hi-fi visuals do not include a border when the cards are not hovered.

jeff-phillips-18 commented on src/styles/variables.less:
```diff
@@ -4,3 +4,6 @@
 @sidebar-os-bg: #30363d;
+
+@saas-text: @color-pf-black-800; //to be replaced with @color-pf-white when the background image is in place
+@saas-hover: fade(#09405e, 80%);
```
We should stick to the Patternfly color palette. @color-pf-blue-600 is pretty close.

spadgett commented on src/styles/saas-list.less:
```diff
@@ -1,53 +1,93 @@
-    cursor: pointer;
-    margin: 0;
+    .card-content {
+      cursor: pointer;
```
This is often the sign of an accessibility problem. Not something for you to change @cshinn, but if the cards aren't links, you won't be able to focus them with a keyboard or screenreader. We should look at this as a follow on.

jwforres replied:
Yep, my suggestion would be that the name on the tile be an `<a>` tag.  Then if we want to style it to not look like an `<a>` we can. I'm opening https://github.com/openshift/origin-web-catalog/issues/67 since this is an issue with more than just the SaaS tiles.

spadgett commented on src/styles/saas-list.less:
```diff
@@ -1,53 +1,93 @@
-      margin-bottom: 20px;
+    .card {
+      background: none;
+      box-shadow: none;
```
It doesn't feel like we should need to set this. Is this coming from `card-view-pf`? Can we just leave that class off since they're no longer styled as cards?

spadgett commented on src/styles/saas-list.less:
```diff
@@ -1,53 +1,93 @@
+      }
+
+      .card {
+        border: none;
```
Same question here: should we just remove the `card-view-pf` class and style the cards ourselves?

cshinn replied:
Definitely, I'll get rid of that.

spadgett commented on src/styles/saas-list.less:
```diff
@@ -56,21 +96,45 @@
+          width: 100%;
+          margin-bottom: 20px;
+
+          &.fa.fa-mobile {
```
We shouldn't be styling specific icon classes as these are customizable.

cshinn replied:
@spadgett will these be font icons at all? I was under the impression that they would probably be images at some point in the future.

spadgett replied:
I'm not sure. We generally try to use font icons when we can.

cc @jwforres

serenamarie125 replied:
@spadgett @cshinn I can't imagine that these will be font icons.  My guess is that it will be some custom icons or product imagery, but I will check on our side.  I think we will document that we want them to be monochrome though so that the color can still be changed?

rhamilto replied:
My recommendation is to use SVG images as I believe the icons in the mocks are part of the [standard brand icons](http://brand.redhat.com/elements/icons/), which are in SVG format, but not available as a font (which is really no longer needed for our purposes since our supported browsers support SVG).

rhamilto commented:
@cshinn, we try to follow [PatternFly's code guide](http://codeguide.patternfly.org/) for html and css.  Of importance here is [css declaration order](http://codeguide.patternfly.org/#css-declaration-order).  Mind making those adjustments and squashing?  It's much easier to review all the changes at once rather than multiple commits?

cshinn commented:
@spadgett I've stripped out references to pf-card-view and re-styled. Much simpler now.

spadgett commented on src/components/saas-list/saas-list.html:
```diff
@@ -1,13 +1,11 @@
-    <div class="card-container">
-      <div class="card-header">
+  <div class="saas-list" items="$ctrl.saasOfferings">
+    <div class="card" ng-repeat="item in $ctrl.saasOfferings" ng-class="{ 'last': $ctrl.saasOfferings.length == ($index+1)}">
```
```
ng-class="{ last: $last }"
```

spadgett commented:
We'll need to update the tests since the markup changed, cc @dtaylor113

spadgett commented:
Even with my browser fullscreen on a 15" laptop, the cards take the entire page.

@serenamarie125 How should this look at this screen width?

<img width="1440" alt="screen shot 2017-03-31 at 1 02 51 pm" src="https://cloud.githubusercontent.com/assets/1167259/24560871/7639378a-1612-11e7-9879-39510817e392.png">

cshinn commented:
@serenamarie125 @spadgett I've reduced the padding, icon size, and title size to get it to fit into a single row on my smaller screen, but without reducing the base font size we're looking at something like this. Seems a bit cramped to me.
<img width="1440" alt="screen shot 2017-03-31 at 1 17 03 pm 2" src="https://cloud.githubusercontent.com/assets/3300756/24561386/932720a8-1614-11e7-9930-c8cf8f2b7802.png">

spadgett commented:
The number of items isn't fixed either. With three, the current stylings are fine. With five, even the smaller fonts and margins would wrap :/

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
-    cursor: pointer;
-    margin: 0;
+  .saas-list {
+    display: flex;
```
Since you don't need flex to achieve your layout for mobile, it's a bit simpler to only do the flex-ing inside of the media query below (line 26).  That way, there is no need to set and reset flex-direction.  In other words, move lines 20, 22, 23 to line 27.  Delete line 21 and 27.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+    color: @saas-text;
+    margin: 0;
+    padding: 20px;
+    padding-left: (@grid-gutter-width / 2);
```
No need to set padding-left to 20px a second time.  :)

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+  h1 {
+    color: @saas-text;
+    margin: 0;
+    padding: 20px;
```
Use (@grid-gutter-width / 2) instead?

dtaylor113 replied:
Hi @rhamilto, just fyi that I initially created this .less file which is why it probably needs some cleaning up :-)  @cshinn is not to blame!!! :-)

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+    margin: 0;
+    padding: 20px;
+    padding-left: (@grid-gutter-width / 2);
+    width: 100%;
```
Not necessary.  Block level elements assuming a width of 100% by default.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
 
-  .card-container {
-    padding: 6px;
+    .last {
```
Use &[:last-child](https://developer.mozilla.org/en-US/docs/Web/CSS/:last-child) on line 44 instead and remove the class="last" from the markup.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
-    text-align: inherit;
-    margin-bottom: 8px;
+    .card {
+      align-items: center;
```
Not needed.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+      cursor: pointer;
+      display: flex;
+      height: 5em;
+      width: 100%;
```
Not needed.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+      @media (min-width: @screen-sm-min) {
+        border-bottom: none;
+        flex: 0 1 auto;
+        height: 300px;
```
One of the benefits of flexbox is that siblings with automatically assuming the same height as one another.  So it's probably best if you don't set a fixed height.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
-        vertical-align: middle;
-        display: table-cell;
-        padding-left: 8px;
+        font-size: @font-size-base + 4;
```
Perhaps this should be scaled down for resolutions of maybe 480 and less so the text isn't so crowded?

![screen shot 2017-03-31 at 3 24 17 pm](https://cloud.githubusercontent.com/assets/895728/24566161/6ecd35aa-1626-11e7-9293-74d8dd3a4046.PNG)

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
-        display: table-cell;
-        padding-left: 8px;
+        font-size: @font-size-base + 4;
+        font-weight: bold;
```
Use a value of [700](https://github.com/patternfly/patternfly/blob/master/src/less/fonts.less#L80) instead of the keyword.

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
+        flex: 0 1 auto;
+        height: 300px;
+        padding: 30px;
+        width: 280px;
```
This fixed width could be problematic when resizing the viewport.  It'd be better (and less fragile) if you made the width relative (percentage-based).

serenamarie125 commented:
I like the updates @cshinn 
For Tech Preview, i think the max we will have is 3.  @spadgett are you saying it looks ok with < 5?

rhamilto commented on src/styles/saas-list.less:
```diff
@@ -1,76 +1,104 @@
 
       @media (min-width: @screen-sm-min) {
-        height: 70px;
+        display: block;
```
Use flex-direction: column instead so you keep things flexed.

cshinn commented:
@rhamilto I appreciate your taking the time to help me improve this stuff! I'll get right on it

cshinn commented:
@spadgett I've updated based on @rhamilto 's feedback and removed the saas header per @dtaylor113 's instructions. Between the visual tweaks and the removal of the header and the scrolling buttons, the catalog section is now visible even when the saas items wrap to a second line.

cshinn commented:
@rhamilto @jwforres I've updated the PR to include the correct text color and added the background placeholder color

cshinn commented:
@rhamilto I've added in the background color as making your suggested changes

jeff-phillips-18 commented:
@rhamilto Any further issues with this?

rhamilto commented on src/styles/variables.less:
```diff
@@ -7,3 +7,6 @@
 @landing-side-bar-width-sm:                           325px;
 @landing-side-bar-width-md:                           375px;
 @landing-side-bar-width-lg:                           450px;
+@landing-main-bg: #3ca2c9;
```
It would be great if you alphabetized these.

rhamilto commented:
There are a few responsive display issues I think should be addressed.

At small viewport sizes, the card contents are not vertically centered:

<img width="464" alt="screen shot 2017-04-06 at 8 07 27 am" src="https://cloud.githubusercontent.com/assets/895728/24753510/6fac575c-1aa0-11e7-861b-8b0c130afcdf.PNG">

At medium-ish viewports, 3 over 1 cards feels a bit odd:

<img width="797" alt="screen shot 2017-04-06 at 8 07 43 am" src="https://cloud.githubusercontent.com/assets/895728/24753542/98943702-1aa0-11e7-9957-326420ce6f83.PNG">

Given we're likely to launch with just 2 cards, I eliminated 2 and noticed the hover state on the second item reveals uneven padding.

<img width="962" alt="screen shot 2017-04-06 at 8 08 24 am" src="https://cloud.githubusercontent.com/assets/895728/24753615/d87f163e-1aa0-11e7-9b7b-7926301c857b.PNG">

cshinn commented:
@rhamilto I just checked and was unable to repro the alignment and hover issues in any browser. 
<img width="476" alt="screen shot 2017-04-06 at 8 17 53 am" src="https://cloud.githubusercontent.com/assets/3300756/24753859/0a1cc15e-1aa2-11e7-84a7-9a22a81f5863.png">
<img width="967" alt="screen shot 2017-04-06 at 8 19 21 am" src="https://cloud.githubusercontent.com/assets/3300756/24753863/12767ef8-1aa2-11e7-9201-528e793c3481.png">

As for centering the bottom item, I checked with a few people and we seemed to prefer it over left alignment (especially with fewer items) but if people feel strongly, I'm happy to change these to left-justified

rhamilto commented:
Yup.  You're correct, @cshinn.  Out of date dependencies was to blame for the responsive issues I was seeing.

spadgett commented:
We should make it look OK without descriptions since these are customizable, but OK as a follow on issue.
